src/ConnectKafka.cpp

```diff
@@ -3,22 +3,39 @@
 #include <iomanip>
 
 namespace {
+
+void logError(const std::string &ErrStr,
+              std::shared_ptr<spdlog::logger> Logger) {
+  if (!ErrStr.empty())
+    Logger->error(ErrStr);
+}
+
 std::unique_ptr<RdKafka::Conf>
 createGlobalConfiguration(const std::string &BrokerAddr) {
   auto conf = std::unique_ptr<RdKafka::Conf>(
       RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL));
   std::string ErrStr;
+  std::shared_ptr<spdlog::logger> Logger = spdlog::get("LOG");
   conf->set("metadata.broker.list", BrokerAddr, ErrStr);
+  logError(ErrStr, Logger);
```

Comment:
Lots of repetition, this is a sign that this should be refactored into a separate function.